File: sdks/js/src/opspec/interpreter/opcall/params/param/object/validate.ts
import Ajv from 'ajv'
const ajv = new Ajv()

/**
 * validates value against constraints
 * @param {Object} value
 * @param {Object} [constraints]
 * @return {Array<Error>}
 */
export default function validate (
  value?: object | string,
  constraints?: any
) {
  constraints = Object.assign({type: 'object'}, constraints)

  ajv.validate(constraints, value)
  return ajv.errors || []
}

File: sdks/go/opspec/interpreter/call/op/params/param/str/validate.go
package str

import (
	"github.com/opctl/opctl/sdks/go/data/coerce"
	"github.com/opctl/opctl/sdks/go/internal/jsonschema"
	"github.com/opctl/opctl/sdks/go/model"
)

// Validate validates a value against a number parameter
func Validate(
	value *model.Value,
	constraints map[string]interface{},
) []error {
	valueAsString, err := coerce.ToString(value)
	if err != nil {
		return []error{err}
	}

	return jsonschema.Validate(
		*valueAsString.String,
		constraints,
	)
}
